apps/web/src/lib/components/ui/community-details/agentLibraryDetail.svelte

```diff
@@ -0,0 +1,107 @@
+<script lang="ts">
+	import * as Dialog from '$lib/components/ui/dialog';
+	import { User } from 'lucide-svelte';
+	import type { Agent } from '$lib/types/models';
+	import { createEventDispatcher } from 'svelte';
+
+	export let displayedAgent: Agent;
+	export let showDetails: boolean;
+	const dispatch = createEventDispatcher();
+
+	function closeDialog() {
+		dispatch('close');
+		showDetails = false
+		console.log(showDetails);
+	}
+
+	function timeSince(dateIsoString: string | number | Date) {
+		const date = new Date(dateIsoString);
+		const now = new Date();
+		const diffInSeconds = Math.round((now - date) / 1000);
+
+		if (diffInSeconds < 60) {
+			return 'just now';
+		} else if (diffInSeconds < 3600) {
+			return `${Math.floor(diffInSeconds / 60)} minute${Math.floor(diffInSeconds / 60) === 1 ? '' : 's'} ago`;
+		} else if (diffInSeconds < 86400) {
+			return `${Math.floor(diffInSeconds / 3600)} hour${Math.floor(diffInSeconds / 3600) === 1 ? '' : 's'} ago`;
+		} else if (diffInSeconds < 172800) {
+			return 'yesterday';
+		} else if (diffInSeconds < 2592000) {
+			return `${Math.floor(diffInSeconds / 86400)} day${Math.floor(diffInSeconds / 86400) === 1 ? '' : 's'} ago`;
+		} else if (diffInSeconds < 31104000) {
+			const months = Math.floor(diffInSeconds / 2592000);
+			if ([1, 2, 3, 6].includes(months)) {
+				return `${months} month${months === 1 ? '' : 's'} ago`;
+			}
+			return `${months} months ago`;
+		} else {
+			const years = Math.floor(diffInSeconds / 31104000);
+			return `${years} year${years === 1 ? '' : 's'} ago`;
+		}
+	}
+</script>
+
+<div class="mx-auto w-full max-w-6xl">
+	<Dialog.Root open={showDetails} onOpenChange={() => closeDialog()}>
```

Comment:
If you want the dialog open state to be controlled by `showDetails`, it should be `onOpenChange={(open) => (showDetails = open)}>`

Reply:
no i don't want it just to open , i am using a dispatch so that the showDetails will also be updated on the parent